Modules/_io/bufferedio.c

```diff
@@ -1795,8 +1795,8 @@ _io_BufferedWriter___init___impl(buffered *self, PyObject *raw,
     _bufferedwriter_reset_buf(self);
     self->pos = 0;
 
-    self->fast_closed_checks = (Py_TYPE(self) == &PyBufferedWriter_Type &&
-                                Py_TYPE(raw) == &PyFileIO_Type);
+    self->fast_closed_checks = Py_IS_TYPE(self, &PyBufferedWriter_Type) &&
+                               Py_IS_TYPE(raw, &PyFileIO_Type);
```

Comment:
Please keep surrounding parenthesis:

```suggestion
    self->fast_closed_checks = (Py_IS_TYPE(self, &PyBufferedWriter_Type) &&
                                Py_IS_TYPE(raw, &PyFileIO_Type));
```